src/Raven.Client/Documents/Smuggler/DatabaseSmuggler.cs

```diff
@@ -49,8 +51,15 @@ public Task<Operation> ExportAsync(DatabaseSmugglerExportOptions options, string
         {
             return ExportAsync(options, async stream =>
             {
-                using (var file = File.OpenWrite(toFile))
-                    await stream.CopyToAsync(file, 8192, token).ConfigureAwait(false);
+                var directoryName = Path.GetDirectoryName(toFile);
+                if (Directory.Exists(directoryName) == false)
+                    Directory.CreateDirectory(directoryName);
+
+                if (Directory.Exists(toFile))
```

Comment:
Do not override the will of the user. Better to do:
```
                var fileInfo = new FileInfo(toFile);
                var directoryInfo = fileInfo.Directory;
                if (directoryInfo != null && directoryInfo.Exists == false)
                    directoryInfo.Create();

                using (var file = fileInfo.OpenWrite())
                    await stream.CopyToAsync(file, 8192, token).ConfigureAwait(false);
```

Reply:
this will throw if last part of the path is existing folder

Reply:
I really hope so, because argument name is 'toFile'. You can detect that situation and throw different exception, for sure do not generate random file name.

Reply:
done